File: win/directx/sdk-9.0/Samples/C#/DirectSound/Play3DSound/Algorithm.cs
//----------------------------------------------------------------------------
// File: Algorithm.cs
//
// Copyright (c) Microsoft Corp. All rights reserved.
//-----------------------------------------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;
using Microsoft.DirectX.DirectSound;

public class AlgorithmForm : Form
{
	private Button buttonOk;
	private Button buttonCancel;
	private RadioButton radiobuttonNoVirtRadio;
	private RadioButton radiobuttonHighVirtRadio;
	private RadioButton radiobuttonLightVirtRadio;

	public AlgorithmForm()
	{
		//
		// Required for Windows Form Designer support
		//
		InitializeComponent();
	}
    #region InitializeComponent code
	private void InitializeComponent()
	{
		this.buttonOk = new System.Windows.Forms.Button();
		this.buttonCancel = new System.Windows.Forms.Button();
		this.radiobuttonNoVirtRadio = new System.Windows.Forms.RadioButton();
		this.radiobuttonHighVirtRadio = new System.Windows.Forms.RadioButton();
		this.radiobuttonLightVirtRadio = new System.Windows.Forms.RadioButton();
		this.SuspendLayout();
		// 
		// buttonOk
		// 
		this.buttonOk.DialogResult = System.Windows.Forms.DialogResult.OK;
		this.buttonOk.Location = new System.Drawing.Point(294, 72);
		this.buttonOk.Name = "buttonOk";
		this.buttonOk.TabIndex = 0;
		this.buttonOk.Text = "OK";
		this.buttonOk.Click += new System.EventHandler(this.buttonOk_Click);
		// 
		// buttonCancel
		// 
		this.buttonCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel;
		this.buttonCancel.Location = new System.Drawing.Point(375, 72);
		this.buttonCancel.Name = "buttonCancel";
		this.buttonCancel.TabIndex = 1;
		this.buttonCancel.Text = "Cancel";
		// 
		// radiobuttonNoVirtRadio
		// 
		this.radiobuttonNoVirtRadio.Checked = true;
		this.radiobuttonNoVirtRadio.Location = new System.Drawing.Point(10, 11);
		this.radiobuttonNoVirtRadio.Name = "radiobuttonNoVirtRadio";
		this.radiobuttonNoVirtRadio.Size = new System.Drawing.Size(339, 16);
		this.radiobuttonNoVirtRadio.TabIndex = 2;
		this.radiobuttonNoVirtRadio.TabStop = true;
		this.radiobuttonNoVirtRadio.Text = "&No Virtualization (WDM or VxD. CPU efficient, but basic 3-D effect)";
		// 
		// radiobuttonHighVirtRadio
		// 
		this.radiobuttonHighVirtRadio.Location = new System.Drawing.Point(10, 28);
		this.radiobuttonHighVirtRadio.Name = "radiobuttonHighVirtRadio";
		this.radiobuttonHighVirtRadio.Size = new System.Drawing.Size(391, 16);
		this.radiobuttonHighVirtRadio.TabIndex = 3;
		this.radiobuttonHighVirtRadio.Text = "&High Quality (WDM only.  Highest quality 3D audio effect, but uses more CPU)";
		// 
		// radiobuttonLightVirtRadio
		// 
		this.radiobuttonLightVirtRadio.Location = new System.Drawing.Point(10, 46);
		this.radiobuttonLightVirtRadio.Name = "radiobuttonLightVirtRadio";
		this.radiobuttonLightVirtRadio.Size = new System.Drawing.Size(430, 16);
		this.radiobuttonLightVirtRadio.TabIndex = 4;
		this.radiobuttonLightVirtRadio.Text = "&Light Quality (WDM only.  Good 3-D audio effect, but uses less CPU than High Qua" +
			"lity)";
		// 
		// AlgorithmForm
		// 
		this.AcceptButton = this.buttonOk;
		this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
		this.ClientSize = new System.Drawing.Size(460, 111);
		this.Controls.AddRange(new System.Windows.Forms.Control[] {
																	  this.buttonOk,
																	  this.buttonCancel,
																	  this.radiobuttonNoVirtRadio,
																	  this.radiobuttonHighVirtRadio,
																	  this.radiobuttonLightVirtRadio});
		this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
		this.MaximizeBox = false;
		this.MinimizeBox = false;
		this.Name = "AlgorithmForm";
		this.Text = "Select 3D Algorithm ";
		this.ResumeLayout(false);

	}
    #endregion

	private void buttonOk_Click(object sender, System.EventArgs e)
	{
		if (true == radiobuttonNoVirtRadio.Checked)
			Play3DSound.guid3DAlgorithm = DSoundHelper.Guid3DAlgorithmNoVirtualization;
		else if (true == radiobuttonHighVirtRadio.Checked)
			Play3DSound.guid3DAlgorithm = DSoundHelper.Guid3DAlgorithmHrtfFull;
		else if (true == radiobuttonLightVirtRadio.Checked)
			Play3DSound.guid3DAlgorithm = DSoundHelper.Guid3DAlgorithmHrtfLight;

		this.Close();
	}
}

File: win/directx/sdk-9.0/Samples/VB.Net/DirectSound/Play3DSound/Algorithm.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports Microsoft.DirectX.DirectSound

Public Class AlgorithmForm
    Inherits Form
    Private WithEvents buttonOk As Button
    Private WithEvents buttonCancel As Button
    Private WithEvents radiobuttonNoVirtRadio As RadioButton
    Private WithEvents radiobuttonHighVirtRadio As RadioButton
    Private WithEvents radiobuttonLightVirtRadio As RadioButton

    Public Sub New()
        '
        ' Required for Windows Form Designer support
        '
        InitializeComponent()
    End Sub 'New

    Sub InitializeComponent()
        Me.buttonOk = New System.Windows.Forms.Button()
        Me.buttonCancel = New System.Windows.Forms.Button()
        Me.radiobuttonNoVirtRadio = New System.Windows.Forms.RadioButton()
        Me.radiobuttonHighVirtRadio = New System.Windows.Forms.RadioButton()
        Me.radiobuttonLightVirtRadio = New System.Windows.Forms.RadioButton()
        Me.SuspendLayout()
        ' 
        ' buttonOk
        ' 
        Me.buttonOk.DialogResult = System.Windows.Forms.DialogResult.OK
        Me.buttonOk.Location = New System.Drawing.Point(294, 72)
        Me.buttonOk.Name = "buttonOk"
        Me.buttonOk.TabIndex = 0
        Me.buttonOk.Text = "OK"
        ' 
        ' buttonCancel
        ' 
        Me.buttonCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel
        Me.buttonCancel.Location = New System.Drawing.Point(375, 72)
        Me.buttonCancel.Name = "buttonCancel"
        Me.buttonCancel.TabIndex = 1
        Me.buttonCancel.Text = "Cancel"
        ' 
        ' radiobuttonNoVirtRadio
        ' 
        Me.radiobuttonNoVirtRadio.Checked = True
        Me.radiobuttonNoVirtRadio.Location = New System.Drawing.Point(10, 11)
        Me.radiobuttonNoVirtRadio.Name = "radiobuttonNoVirtRadio"
        Me.radiobuttonNoVirtRadio.Size = New System.Drawing.Size(339, 16)
        Me.radiobuttonNoVirtRadio.TabIndex = 2
        Me.radiobuttonNoVirtRadio.TabStop = True
        Me.radiobuttonNoVirtRadio.Text = "&No Virtualization (WDM or VxD. CPU efficient, but basic 3-D effect)"
        ' 
        ' radiobuttonHighVirtRadio
        ' 
        Me.radiobuttonHighVirtRadio.Location = New System.Drawing.Point(10, 28)
        Me.radiobuttonHighVirtRadio.Name = "radiobuttonHighVirtRadio"
        Me.radiobuttonHighVirtRadio.Size = New System.Drawing.Size(391, 16)
        Me.radiobuttonHighVirtRadio.TabIndex = 3
        Me.radiobuttonHighVirtRadio.Text = "&High Quality (WDM only.  Highest quality 3D audio effect, but uses more CPU)"
        ' 
        ' radiobuttonLightVirtRadio
        ' 
        Me.radiobuttonLightVirtRadio.Location = New System.Drawing.Point(10, 46)
        Me.radiobuttonLightVirtRadio.Name = "radiobuttonLightVirtRadio"
        Me.radiobuttonLightVirtRadio.Size = New System.Drawing.Size(430, 16)
        Me.radiobuttonLightVirtRadio.TabIndex = 4
        Me.radiobuttonLightVirtRadio.Text = "&Light Quality (WDM only.  Good 3-D audio effect, but uses less CPU than High Qua" + "lity)"
        ' 
        ' AlgorithmForm
        ' 
        Me.AcceptButton = Me.buttonOk
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.ClientSize = New System.Drawing.Size(460, 111)
        Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.buttonOk, Me.buttonCancel, Me.radiobuttonNoVirtRadio, Me.radiobuttonHighVirtRadio, Me.radiobuttonLightVirtRadio})
        Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
        Me.MaximizeBox = False
        Me.MinimizeBox = False
        Me.Name = "AlgorithmForm"
        Me.Text = "Select 3D Algorithm "
        Me.ResumeLayout(False)
    End Sub 'InitializeComponent

    Private Sub buttonOk_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles buttonOk.Click
        If True = radiobuttonNoVirtRadio.Checked Then
            Play3DSound.guid3DAlgorithm = DSoundHelper.Guid3DAlgorithmNoVirtualization
        ElseIf True = radiobuttonHighVirtRadio.Checked Then
            Play3DSound.guid3DAlgorithm = DSoundHelper.Guid3DAlgorithmHrtfFull
        ElseIf True = radiobuttonLightVirtRadio.Checked Then
            Play3DSound.guid3DAlgorithm = DSoundHelper.Guid3DAlgorithmHrtfLight
        End If
        Me.Close()
    End Sub 'buttonOk_Click
End Class 'AlgorithmForm